addon/-private/system/store.js

```diff
@@ -2065,32 +2047,53 @@ Store = Service.extend({
     @return {DS.Model}
   */
   modelFor(modelName) {
-    heimdall.increment(modelFor);
     assert("You need to pass a model name to the store's modelFor method", isPresent(modelName));
     assert('Passing classes to store methods has been removed. Please pass a dasherized string instead of '+ inspect(modelName), typeof modelName === 'string');
 
-    let factory = this.modelFactoryFor(modelName);
-    if (!factory) {
-      //Support looking up mixins as base types for polymorphic relationships
-      factory = this._modelForMixin(modelName);
-    }
+    let trueModelName = this._classKeyFor(modelName);
+
+    return this._modelFor(trueModelName);
+  },
+
+  /*
+    @private
+   */
+  _modelFor(modelName) {
```

Comment:
FTR, the reason we can't remove `modelFactoryFor` at the moment is because `adapters` use it to do a `modelFor` lookup without risk of erroring when a model isn't found when an unknown type is encountered in a payload (prints a warning, doesn't error).